run/dpl_eventgen.cxx

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include "../Framework/Core/src/ArrowSupport.h"
+#include "Framework/AnalysisTask.h"
+#include "Monitoring/Monitoring.h"
+#include "Framework/CommonDataProcessors.h"
+#include "SimulationDataFormat/MCTrack.h"
+#include "Framework/runDataProcessing.h"
+#include <Generators/GeneratorService.h>
+#include <CommonUtils/ConfigurableParam.h>
+#include <CommonUtils/RngHelper.h>
+
+using namespace o2::framework;
+
+struct GeneratorTask {
+  Configurable<std::string> params{"confKeyValues", "", "configurable params - configuring event generation internals"};
+  Configurable<std::string> generator{"generator", "boxgen", "Name of generator"};
+  Configurable<std::string> trigger{"trigger", "", "Trigger"}; // <---- to be treated
+  Configurable<int> eventNum{"nevents", 1, "Number of events"};
+  Configurable<long> seed{"seed", 0, "(TRandom) Seed"};
+  Configurable<int> aggregate{"aggregate-timeframe", 300, "Number of events to put in a timeframe"};
+  Configurable<std::string> vtxModeArg{"vertexMode", "kDiamondParam", "Where the beam-spot vertex should come from. Must be one of kNoVertex, kDiamondParam, kCCDB"};
```

Comment:
As I understand, `SimConfigData` has a ROOT dictionary and only has simple types. Hence it can be used as a configurable, somethin lilke: `Configurable<o2::conf::SimConfigData> simconfig{"sim-config", SimConfigData{}, "simulation config"};`. It can be then set via JSON or overridden from `confKeyValues`.

Reply:
SimConfigData is not by itself a configurable param so I don't think it can be influenced via confKeyValues.

Reply:
What I meant is that values, passed through the other configurable, `params` can override whatever is set through SimConfigData. Specifically for hyperloop, though, each element of SimConfigData would be adjustable through the interface because configurable structs are already supported.